sway/focus.c

```diff
@@ -140,33 +140,32 @@ bool set_focused_container(swayc_t *c) {
 	}
 
 	// get new focused view and set focus to it.
-	p = get_focused_view(c);
-	if (p->type == C_VIEW && !(wlc_view_get_type(p->handle) & WLC_BIT_POPUP)) {
+	if (c->type == C_CONTAINER || (c->type == C_VIEW && !(wlc_view_get_type(p->handle) & WLC_BIT_POPUP))) {
 		// unactivate previous focus
 		if (focused->type == C_VIEW) {
 			wlc_view_set_state(focused->handle, WLC_BIT_ACTIVATED, false);
-			update_view_border(focused);
 		}
+		update_container_border(focused);
```

Comment:
Does this need a guard for if view focused is locked? (I'm not sure when view focus can be locked).

Also, does the order that we do things here matter much? It could be a bit simpler if not.